Custom authenticator main OKTA-508649 (#3491)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/push-providers/index.md

---
title: Push Providers
category: management
---

# Push Providers API

<ApiLifecycle access="ea" /></br>
<ApiLifecycle access="ie" />

> **Note:** This feature is only available as a part of Okta Identity Engine. Contact your Okta account team or ask us on our [forum](https://devforum.okta.com/) for further information.

The Okta Push Providers API provides a centralized integration platform to fetch and manage push provider configurations. Okta administrators can use these APIs to provide their push provider credentials, for example from APNs and FCM, so that Okta can send push notifications to their own custom app authenticator applications.

> The Push Providers API supports the following **Authorization Schemes**:

Reviewer: why is Authorization Schemes in bold?

> The Push Providers API supports the following **Authorization Schemes**:

Author: ```suggestion
The Push Providers API supports the following Authorization Schemes:
```

-H "Authorization: Bearer ${oauth_token}" \
-d '{
  "providerType": "APNS",
  "name": "Example Push Provider 1",
  "configuration": {
    "keyId": "KEY_ID",
    "teamId": "TEAM_ID",
    "tokenSigningKey": "-----BEGIN PRIVATE KEY-----\nPRIVATE_KEY\n-----END PRIVATE KEY-----\n",
    "fileName": "fileName.p8"
  }
}' "https://${yourOktaDomain}/api/v1/push-providers"
```

##### Response

```json
{
  "id": "ppchvbeucdTgqeiGxR0g4",
  "providerType": "APNS",
  "name": "Example Push Provider 1",
  "lastUpdatedDate": "2022-01-00T00:00:00.000Z",
  "configuration": {
    "keyId": "KEY_ID",
    "teamId": "TEAM_ID",
    "fileName": "fileName.p8"
  },
  "_links": {
    "self": {
      "href": "https://{yourOktaDomain}}/api/v1/push-providers/{pushProviderId}",
      "hints": {
        "allow": [ "GET", "PUT", "DELETE"]
      }
    }
  }
}
```

### List Push Providers

<ApiOperation method="get" url="/api/v1/push-providers" />

Lists all available Push Providers

#### Permitted OAuth 2.0 scopes
`okta.pushProviders.read`

#### Request path parameters

None

#### Request query parameters

| Parameter | Type   | Description | Required |
| --------- | ------ | -------- | ---- |

> | `type`    |  String | Returns a list of Push Providers that match the specific `providerType` in the [Push Provider object](#push-provider-object). One of: `APNS`, `FCM` | FALSE |

Reviewer: ```suggestion
| `type`    |  String | Returns a list of Push Providers that match the specific `providerType` in the [Push Provider object](#push-provider-object). Supported values: `APNS` or `FCM` | FALSE |
```

-H "Authorization: Bearer ${oauth_token}" \
"https://${yourOktaDomain}/api/v1/push-providers"
```

##### Response
```json
[
  {
    "id": "ppchvbeucdTgqeiGxR0g4",
    "providerType": "APNS",
    "name": "Example Push Provider 1",
    "lastUpdatedDate": "2022-01-00T00:00:00.000Z",
    "configuration": {
      "keyId": "ABC123DEFG",
      "teamId": "DEF123GHIJ",
      "fileName": "fileName.p8"
    },
    "_links": {
      "self": {
        "href": "https://{yourOktaDomain}}/api/v1/push-providers/{pushProviderId}",
        "hints": {
          "allow": [ "GET", "PUT", "DELETE"]
        }
      }
    }
  },
  {
    "id": "ppctekcmngGaqeiBxB0g4",
    "providerType": "FCM",
    "name": "Example Push Provider 2",
    "lastUpdatedDate": "2022-01-00T00:00:00.000Z",
    "configuration": {
      "projectId": "PROJECT_ID",
      "fileName": "fileName.json",
    },
    "_links": {
      "self": {
        "href": "https://{yourOktaDomain}}/api/v1/push-providers/{pushProviderId}",
        "hints": {
          "allow": [ "GET", "PUT", "DELETE"]
        }
      }
    }
  }
]
```

### Get Push Provider by ID

<ApiOperation method="get" url="/api/v1/push-providers/${pushProviderId}" />

> Fetches a Push Provider by its `id`. If you do not know the `id`, you can [List Push Providers](#list-push-providers).

Reviewer: ```suggestion
Fetches a Push Provider by its `id`. If you don't know the `id`, you can [List Push Providers](#list-push-providers).
```

HTTP/1.1 409 Conflict
Content-Type: application/json
```
```json
{
  "errorCode": "E0000187",
  "errorSummary": "Cannot delete push provider because it is being used by a custom app authenticator.",
  "errorLink": "E0000187",
  "errorId": "oaenwA1ra80S9W-pvbh4m6haA",
  "errorCauses": [
    {
      "errorSummary": "Cannot delete push provider notification service configuration with id={{pushProviderId}} because it is associated with the following custom app authenticator ids=[{{authenticatorId}}]. Please unassociate it from the authenticators and try again."
    }
  ]
}
```

## Push Providers API objects

### Push Provider object

#### Push Provider properties

The Push Provider object has the following properties:

| Property          | Type                                   | Description                                        |
| ----------------- | -------------------------------------- | -------------------------------------------------- |
| `configuration`   | [Configuration](#configuration-object) | Object that contains `providerType` specific configurations |
| `name`            | String                                 | Display name of the Push Provider                  |

> | `providerType`    | String                                 | Provider type for the Push Provider. One of `APNS`, `FCM` |

Reviewer: ```suggestion
| `providerType`    | String                                 | Provider type for the Push Provider. Supported values: `APNS` or `FCM` |
```

#### Configuration object

| Property             | Type            | Description                                        | `providerType` |
| -------------------- | --------------- | -------------------------------------------------- | ---------------------------------------- |
| `fileName`           | String          | (Optional) File name for Admin Console display     | All
| `keyId`              | String          | 10-character Key ID obtained from the Apple developer account                   | `APNS`                                   |

> | `serviceAccountJson` | JSON            | JSON containing the private service account key and service account details. See [Creating and managing service account keys](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) for more information on creating service account keys in JSON.           | `FCM`                                    |

Reviewer: ```suggestion
| `serviceAccountJson` | JSON            | JSON that contains the private service account key and service account details. See [Creating and managing service account keys](https://cloud.google.com/iam/docs/creating-managing-service-account-keys).           | `FCM`                                    |
```

        "client_x509_cert_url": "https://www.googleapis.com/robot/v1/metadata/x509/SERVICE_ACCOUNT_EMAIL"
    },
    "fileName": "fileName.json"
  }
}
```

### Push Provider response object

#### Push Provider response properties

The Push Provider response object has the following properties:

| Property          | Type                                   | Description                                        |
| ----------------- | -------------------------------------- | -------------------------------------------------- |
| `_links`          | [Link](#link-object)                   | Allowed operations for the Push Provider           |
| `configuration`   | [Configuration response](#configuration-response-object) | Object that contains `providerType` specific configurations |
| `id`              | String                                 | Unique key for the Push Provider                   |
| `lastUpdatedDate` | String (ISO-8601)                      | Timestamp when the Push Provider was last modified |
| `name`            | String                                 | Display name of the Push Provider                  |

> | `providerType`    | String                                 | Provider type for the Push Provider. One of `APNS`, `FCM` |

Reviewer: ```suggestion
| `providerType`    | String                                 | Provider type for the Push Provider. Supported values: `APNS` or `FCM` |
```

#### Link object

> For a Push Provider result, the `_links` is read-only and contains a full set of available operations using the [JSON Hypertext Application Language](https://tools.ietf.org/html/draft-kelly-json-hal-06) specification. The `hints` property provides information on allowed HTTP verbs for the `href` property. See [Web Linking](https://tools.ietf.org/html/rfc8288)) for more information on link relations.

Reviewer: ```suggestion
For a Push Provider result, the `_links` is read-only and contains a full set of available operations using the [JSON Hypertext Application Language](https://tools.ietf.org/html/draft-kelly-json-hal-06) specification. The `hints` property provides information on allowed HTTP verbs for the `href` property. See [Web Linking](https://tools.ietf.org/html/rfc8288) for more information on link relations.
```
